src/main/js/field/TableDisplay.js

```diff
@@ -195,28 +221,47 @@ class TableDisplay extends Component {
                 uiValidation: validationState
             });
         };
-        this.props.onConfigTest(callback);
+        onConfigTest(callback);
+    }
+
+    handleCancel() {
+        this.hideModal();
+        this.handleClose();
     }
 
-    createEditModal() {
-        const { currentRowSelected } = this.state;
-        const { modalTitle, newConfigFields, inProgress, saveButton, testButton, testButtonLabel, errorDialogMessage, actionMessage } = this.props;
-        const popupActionMessage = errorDialogMessage ? errorDialogMessage : actionMessage;
+    createTableModal() {
+        const tablePopupRef = this.tablePopup.current;
+        const { currentRowSelected, isInsertModal } = this.state;
+        const {
+            modalTitle, newConfigFields, inProgress, saveButton, testButton, testButtonLabel, errorDialogMessage,
+            actionMessage
+        } = this.props;
+        const popupActionMessage = errorDialogMessage || actionMessage;
+        const configFields = isInsertModal ? newConfigFields() : newConfigFields(currentRowSelected);
+        let cancelFunction = this.handleCancel;
+        if (isInsertModal) {
+            cancelFunction = tablePopupRef && tablePopupRef.onCancel;
+        }
+        let submitFunction = this.handleSubmit;
+        if (isInsertModal) {
+            submitFunction = tablePopupRef && tablePopupRef.handleSubmit;
+        }
+        let testFunction = this.handleTest;
+        if (isInsertModal) {
+            testFunction = tablePopupRef && tablePopupRef.handleTest;
+        }
```

Comment:
Can all of these if checks be condensed into 1?

Reply:
done will update soon.